packages/engine/src/HedronEngine.ts

```diff
@@ -42,21 +43,34 @@ export class HedronEngine {
     this.store.setState({ isSketchModulesReady: true })
   }
 
-  public async addSketchModule(moduleId: string) {
+  public async addSketchModule(moduleId: string): Promise<Result<SketchModuleItem>> {
     if (!this.sketchesUrl) throw new Error('Sketches URL not ready')
 
-    const moduleItem = await importSketchModule(this.sketchesUrl, moduleId)
+    const result = await importSketchModule(this.sketchesUrl, moduleId)
+
+    if (!result.success) {
+      // TODO: check for errors and show UI error
+      return result
+    }
+
+    const moduleItem = result.data
     this.store.getState().setSketchModuleItem(moduleItem)
 
-    return moduleItem
+    return result
   }
 
   public removeSketchModule = async (moduleId: string): Promise<void> => {
     this.store.getState().deleteSketchModule(moduleId)
   }
 
-  public async reimportSketchModuleAndReloadSketches(moduleId: string) {
-    const moduleItem = await this.addSketchModule(moduleId)
+  public async reimportSketchModuleAndReloadSketches(moduleId: string): Promise<void> {
+    const result = await this.addSketchModule(moduleId)
+
+    if (!result.success) {
+      return
+    }
```

Comment:
Because we're now checking for this issue lower down, other functions are affected. Here we're just not doing the reimport stuff if the module failed to load.

Reply:
yeah, this looks really nice, feel free to merge :)